flink-runtime/src/test/java/org/apache/flink/runtime/leaderelection/ZooKeeperLeaderElectionConnectionHandlingTest.java

```diff
@@ -281,6 +282,10 @@ public void notifyLeaderAddress(LeaderInformation leaderInformation) {
 		}
 
 		public CompletableFuture<String> next() {
+			return next(timeout);
```

Comment:
Can we clean up the `timeout` field of the `QueueLeaderElectionListener`? I think it does not make sense that the leader election listener has such a field.

Reply:
Also, `notifyLeaderAddress` should probably wait until it can add the `leaderInformation` to the `queue`.